src/Defra.PTS.Web.Application/Validation/PetMicrochipValidator.cs

```diff
@@ -18,19 +17,24 @@ public PetMicrochipValidator()
                 When(x => !string.IsNullOrWhiteSpace(x.MicrochipNumber), () =>
                 {
                     RuleFor(x => x.MicrochipNumber)
-                        .Length(AppConstants.MaxLength.PetMicrochipNumber)
-                        .WithMessage("Enter your pet’s 15-digit microchip number")
-                        .Unless(x => string.IsNullOrEmpty(x.MicrochipNumber) || x.MicrochipNumber.Length == 15);
-
-                    RuleFor(x => x.MicrochipNumber)
-                        .Matches(@"^\d{15}$")
-                        .WithMessage("Enter a 15-digit number, using only numbers")
-                    .When(x => !string.IsNullOrEmpty(x.MicrochipNumber) && x.MicrochipNumber.Length == 15);
-
-
-
+                        .Custom((microchipNumber, context) =>
+                        {
+                            if (IsAllDigits(microchipNumber) && microchipNumber.Length != 15)
+                            {
+                                context.AddFailure("MicrochipNumber", "Enter your pet’s 15-digit microchip number");
+                            }
+                            else if (!IsAllDigits(microchipNumber) || microchipNumber.Length != 15)
```

Comment:
Does this flag both errors? I suspect getting rid of the second bit of the if might satisfy everything

Reply:
I did it that way before but the test cases didn't pass through so that I added the second bit.

Reply:
Ah cool ok